src/test/kotlin/org/objectionary/deog/unit/graph/TestBase.kt

```diff
@@ -41,46 +44,49 @@ interface TestBase {
     val sep: Char
         get() = File.separatorChar
 
-    /**
-     * Constructs test execution process
-     */
-    fun doTest()
-
     /**
      * Compares expected test output with the actual one
      *
      * @param expected expected output
      * @param actual actual output
+     * @param message message
      */
     @Suppress("KDOC_WITHOUT_RETURN_TAG")
     fun checkOutput(
         expected: String,
-        actual: String
+        actual: String,
+        message: String
     ) =
         assertEquals(
             expected.replace("\n", "").replace("\r", ""),
-            actual.replace("\n", "").replace("\r", "")
+            actual.replace("\n", "").replace("\r", ""),
+            message
         )
 
     /**
      * @param directoryName name of the input directory
      * @return path to input location
      */
-    fun constructInPath(directoryName: String): String = "src${sep}test${sep}resources${sep}unit${sep}in$sep$directoryName"
+    fun constructInPath(directoryName: String): Path =
+        Path.of("src${sep}test${sep}resources${sep}unit${sep}in$sep$directoryName")
 
     /**
      * @param directoryName name of the output directory
      * @return path to output location
      */
-    fun constructOutPath(directoryName: String): String
+    fun constructOutPath(directoryName: String): Path
 
     /**
-     * @return name of the test being executed
+     * Deletes temporary output directory
      *
-     * @todo #34:90min/DEV use @ParameterizedTest here instead of looking into stackTrace. Current approach has some
-     * limitations. For example, if stack depth will increase, this function will output wrong name.
+     * @param path path to source directory
+     * @throws IOException
      */
-    fun getTestName() = Thread.currentThread().stackTrace[4].methodName
-        .substring(5)
-        .replace(' ', '_')
+    fun deleteTempDir(path: Path) {
+        try {
+            FileUtils.deleteDirectory(path.toFile())
+        } catch (e: IOException) {
+            throw IOException("Trying to delete not existing temporary directory", e)
```

Comment:
@c71n93 I'm not sure this is a good idea to re-throw the same exception. Maybe we can choose another exception or create a new one?

Reply:
@Graur yeah, it's really not good. I think it's better to get rid of this function and use `FileUtils.deleteDirectory(path.toFile())` in place. In case of an exception, there will be enough context to understand what happened and not create a wrapper over the `IOException`.